File: 0048-rotate-image/0048-rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        int n = matrix.length;
        // Transpose the matrix
        for (int i = 0; i < n; ++i) {
            for (int j = i; j < n; ++j) {
                int t = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = t;
            }
        }
        // Now reverse each row
        for (int i = 0; i < n; ++i) {
            for (int j = 0; j < n / 2; ++j) {
                int t = matrix[i][j];
                matrix[i][j] = matrix[i][n - j - 1];
                matrix[i][n - j - 1] = t;
            }
        }
    }
}

File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    // Basically this ques. asks to raotate a image by 90 deg
    void rotate(vector<vector<int>>& mat) {
        int n = mat.size();
        // Transepose of a matrix first
        for (int i = 0; i < n; ++i) {
            for (int j = i; j < n; ++j)
                swap(mat[i][j], mat[j][i]);
        }

        // Now Reverse Each Row to get desire result
        for (uint i = 0; i < n; ++i) {
            for (uint j = 0; j < (n / 2); ++j)
                swap(mat[i][j], mat[i][n - j - 1]);
        }
    }
};